models/eprop_synapse_iaf_psc_delta_adapt.h

```diff
@@ -0,0 +1,300 @@
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef EPROP_SYNAPSE_IAF_PSC_DELTA_ADAPT_H
+#define EPROP_SYNAPSE_IAF_PSC_DELTA_ADAPT_H
+
+// nestkernel
+#include "connection.h"
+#include "eprop_synapse.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: synapse, e-prop-plasticity
+
+Short description
++++++++++++++++++
+
+Synapse type for e-prop plasticity
+
+Description
++++++++++++
+
+``eprop_synapse`` is a connector to create e-prop synapses between postsynaptic
+neurons :math:`j` and presynaptic neurons and :math:`i` as defined in [1]_.  The
+change of the synaptic weight :math:`\Delta W_{ji}` depends on the presynaptic
+spikes :math:`z_i^{t-1}`, the pseudo-derivative of the postsynaptic membrane
+voltage :math:`\psi_j^t` (which together form the eligibility trace
+:math:`e_{ji}`), and the learning signal :math:`L_j^t` emitted by the readout
+neurons. The eligibility trace and the presynaptic spike trains are
+low-pass-filtered with kernels :math:`\mathcal{F}_\kappa` with
+:math:`\kappa=\exp\left(\frac{-\delta t}{\tau_\text{out}}\right)` and
+:math:`\mathcal{F}_\alpha` with
+:math:`\kappa=\exp\left(\frac{-\delta t}{\tau_\text{m, out}}\right)`.
+The overall weight update is scaled by the learning rate :math:`\eta`. The
+general formula computing weight updates for eprop synapses projecting onto
+recurrent neurons are thus given by:
+
+.. math::
+  \Delta W_{ji}^\text{rec} &= -\eta \sum_t L_j^t \bar{e}_{ji}^t \\
+   &= -\eta \sum_t L_j^t \mathcal{F}_\kappa \left( \psi^t_j \bar{z}_i^{t-1}\right) \\
+   &= -\eta \sum_t L_j^t \sum_{t'\leq t} \kappa^{t-t'} \psi^t_j \mathcal{F}_\alpha\left( z_i^{t-1}\right)\,.
+
+If the postsynaptic neuron is a adaptive neuron, next to the membrane voltage, a
+second hidden state variable, the threshold adaptation, is present, which
+changes the eligibility trace:
+
+.. math::
+  e_{ji}^t &= \psi_j^t \left(\bar{z}^{t-1} - \beta \epsilon_{ji,a}^{t-1}\right)\,, \\
+  \epsilon^{t-1}_{ji,\text{a}} &= \psi_j^{t-1}\bar{z}_i^{t-2} + \left( \rho - \psi_j^{t-1} \beta \right)
+  \epsilon^{t-2}_{ji,a}\,, \\ \rho &= \exp\left(-\frac{\delta t}{\tau_\text{a}}\right)\,.
+
+Furthermore, a firing rate regularization mechanism keeps the average firing
+rate :math:`f^\text{av}_j` of the postsynaptic neuron close to a target firing rate
+:math:`f^\text{target}`.
+
+.. math::
+  \Delta W_{ji}^\text{reg} = \eta c_\text{reg}
+  \sum_t \frac{1}{Tn_\text{trial}} \left( f^\text{target}-f^\text{av}_j\right)e_{ji}^t,
+
+whereby :math:`c_\text{reg}` scales the overall regularization and the average
+is taken over the time that passed since the last update, i.e., the number of
```

Comment:
```suggestion
is taken over the time that passed since the last update, that is, the number of
```

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/a7ca1793f2482991ad15baa19c6c12c83be665b0